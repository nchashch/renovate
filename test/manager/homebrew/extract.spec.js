const fs = require('fs');
const { extractPackageFile } = require('../../../lib/manager/homebrew/extract');

const aalib = fs.readFileSync(
  'test/manager/homebrew/_fixtures/aalib.rb',
  'utf8'
);
const aap = fs.readFileSync('test/manager/homebrew/_fixtures/aap.rb', 'utf8');
const acmetool = fs.readFileSync(
  'test/manager/homebrew/_fixtures/acmetool.rb',
  'utf8'
);
const aide = fs.readFileSync('test/manager/homebrew/_fixtures/aide.rb', 'utf8');
const ibazel = fs.readFileSync(
  'test/manager/homebrew/_fixtures/ibazel.rb',
  'utf8'
);

describe('lib/manager/homebrew/extract', () => {
  describe('extractPackageFile()', () => {
    it('skips sourceforge dependency', () => {
      expect(extractPackageFile(aalib)).not.toBeNull();
      expect(extractPackageFile(aalib).deps[0].skipReason).toBe(
        'unsupported-url'
      );
      expect(extractPackageFile(aalib)).toMatchSnapshot();
    });
    it('skips sourceforge dependency', () => {
      expect(extractPackageFile(aap)).not.toBeNull();
      expect(extractPackageFile(aap).deps[0].skipReason).toBe(
        'unsupported-url'
      );
      expect(extractPackageFile(aap)).toMatchSnapshot();
    });
    it('skips github dependency with wrong format', () => {
      expect(extractPackageFile(acmetool)).not.toBeNull();
      expect(extractPackageFile(acmetool).deps[0].skipReason).toBe(
        'unsupported-url'
      );
      expect(extractPackageFile(acmetool)).toMatchSnapshot();
    });
    it('extracts "releases" github dependency', () => {
      expect(extractPackageFile(aide)).not.toBeNull();
      expect(extractPackageFile(aide).deps[0].skipReason).toBeUndefined();
      expect(extractPackageFile(aide)).toMatchSnapshot();
    });
    it('extracts "archive" github dependency', () => {
      expect(extractPackageFile(ibazel)).not.toBeNull();
      expect(extractPackageFile(ibazel).deps[0].skipReason).toBeUndefined();
      expect(extractPackageFile(ibazel)).toMatchSnapshot();
    });
  });
});
