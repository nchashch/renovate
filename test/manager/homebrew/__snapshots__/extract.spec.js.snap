// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/manager/homebrew/extract extractPackageFile() extracts "archive" github dependency 1`] = `
Object {
  "deps": Array [
    Object {
      "currentValue": "v0.8.2",
      "depName": "Ibazel",
      "sha256": "26f5125218fad2741d3caf937b02296d803900e5f153f5b1f733f15391b9f9b4",
      "url": "https://github.com/bazelbuild/bazel-watcher/archive/v0.8.2.tar.gz",
    },
  ],
}
`;

exports[`lib/manager/homebrew/extract extractPackageFile() extracts "releases" github dependency 1`] = `
Object {
  "deps": Array [
    Object {
      "currentValue": "v0.16.1",
      "depName": "Aide",
      "sha256": "0f2b7cecc70c1a27d35c06c98804fcdb9f326630de5d035afc447122186010b7",
      "url": "https://github.com/aide/aide/releases/download/v0.16.1/aide-0.16.1.tar.gz",
    },
  ],
}
`;

exports[`lib/manager/homebrew/extract extractPackageFile() skips github dependency with wrong format 1`] = `
Object {
  "deps": Array [
    Object {
      "currentValue": null,
      "depName": "Acmetool",
      "sha256": "6f2cf5cfb987a2df2f791c162209039804fd8fd12692da69f52153ec9668e9ca",
      "skipReason": "unsupported-url",
      "url": "https://github.com/hlandau/acme.git",
    },
  ],
}
`;

exports[`lib/manager/homebrew/extract extractPackageFile() skips sourceforge dependency 1`] = `
Object {
  "deps": Array [
    Object {
      "currentValue": null,
      "depName": "Aalib",
      "sha256": "fbddda9230cf6ee2a4f5706b4b11e2190ae45f5eda1f0409dc4f99b35e0a70ee",
      "skipReason": "unsupported-url",
      "url": "https://downloads.sourceforge.net/aa-project/aalib-1.4rc5.tar.gz",
    },
  ],
}
`;

exports[`lib/manager/homebrew/extract extractPackageFile() skips sourceforge dependency 2`] = `
Object {
  "deps": Array [
    Object {
      "currentValue": null,
      "depName": "Aap",
      "sha256": "3f53b2fc277756042449416150acc477f29de93692944f8a77e8cef285a1efd8",
      "skipReason": "unsupported-url",
      "url": "https://downloads.sourceforge.net/project/a-a-p/aap-1.094.zip",
    },
  ],
}
`;
